runtime/lua/vim/lsp/util.lua

```diff
@@ -1344,6 +1344,10 @@ local function close_preview_window(winnr, bufnrs)
 
     local augroup = 'preview_window_' .. winnr
     pcall(api.nvim_del_augroup_by_name, augroup)
+    local ok, buf = pcall(api.nvim_win_get_var, winnr, 'buf_hold_win')
```

Comment:
Prefer using `vim.w` and `vim.b`.

```suggestion
    local buf = vim.w[winnr].buf_hold_win
    if buf then
      vim.b[buf].lsp_floating_window = nil
    end
```